### PR TITLE
Make dining balance calls async, add no dining plan found page

### PR DESCRIPTION
**Pull request by aantenberg**
Changed dining balance API calls to async, and added dining analytics "no dining plan found" page when a user opens the dining analytics page but do not have a dining plan. Also handling the case where a user has a swipe plan, but no dollar plan, and vice versa, by only showing the graph that has data. (Note: I'm not sure if this is actually possible by the University's meal plan options, but it can't hurt to handle it.)

simulator_screenshot_D584A9A8-511C-43B8-AA84-61EF03F5A270

### CODE REVIEW

File: PennMobile/Dining/SwiftUI/Views/Venue/DiningVenueView.swift [hunk 1]
```diff
@@ -61,7 +59,7 @@ struct CustomHeader: View {
     }
 
     var body: some View {
-        HStack {
+        ZStack {
```
Comment (JHawk0224) [inline PennMobile/Dining/SwiftUI/Views/Venue/DiningVenueView.swift:62]: Shouldn't this stay as an HStack?

File: PennMobile/Dining/SwiftUI/Views/Venue/DiningVenueView.swift [hunk 2]
```diff
@@ -73,7 +71,9 @@ struct CustomHeader: View {
                         showMissingDiningTokenAlert = true
                         return
                     }
-                    DiningViewModelSwiftUI.instance.refreshBalance()
+                    Task.init() {
+                        await DiningViewModelSwiftUI.instance.refreshBalance()
+                    }
```
Comment (01jongmin) [inline PennMobile/Dining/SwiftUI/Views/Venue/DiningVenueView.swift:76]: You can add a `.task` modifier at the end of the view to trigger a async function. More details can be found [here](https://developer.apple.com/documentation/swiftui/view/task(priority:_:))

**Comment by JHawk0224** [comment]
Looks good other than the comments above and the merge conflicts!

**Review by JHawk0224 — APPROVED** [review]
Looks good

**Pull request merged**